File: dart_core/lib/src/receive_code.dart
/// Builds the Telebirr receive code string from order components.
///
/// The receive code is formatted as:
/// `TELEBIRR$BUYGOODS{shortCode}{amount}{prepayId}%{timeout}`
///
/// This code is used in QR-code-based payment flows where the user scans
/// a code to initiate payment.
///
/// [shortCode] is the merchant short code (e.g., `'220311'`).
/// [amount] is the payment amount as a string (e.g., `'100'`).
/// [prepayId] is the prepay ID returned from the create-order API.
/// [timeout] is the payment timeout express value (e.g., `'120m'`).
///
/// Returns the formatted receive code string.
///
/// Example:
/// ```dart
/// final code = buildReceiveCode('220311', '100', 'PREPAY123', '120m');
/// // Returns 'TELEBIRR$BUYGOODS220311100PREPAY123%120m'
/// ```
String buildReceiveCode(
  String shortCode,
  String amount,
  String prepayId,
  String timeout,
) {
  return 'TELEBIRR\$BUYGOODS$shortCode$amount$prepayId%$timeout';
}

File: ts-core/src/receive-code.ts
/**
 * Builds a Telebirr receive code string for the mobile SDK payment flow.
 *
 * Format: `TELEBIRR$BUYGOODS{shortCode}{amount}{prepay_id}%{timeout}`
 *
 * @param shortCode - Merchant short code (e.g. `"220311"`).
 * @param amount - Payment amount in ETB.
 * @param prepayId - Prepay ID returned from the create-order response.
 * @param timeout - Payment timeout expression (e.g. `"120m"`).
 * @returns The formatted receive code string.
 *
 * @example
 * ```ts
 * buildReceiveCode('220311', '100', 'PREPAY123', '120m');
 * // "TELEBIRR$BUYGOODS220311100PREPAY123%120m"
 * ```
 */
export function buildReceiveCode(
  shortCode: string,
  amount: string | number,
  prepayId: string,
  timeout: string,
): string {
  return `TELEBIRR$BUYGOODS${shortCode}${amount}${prepayId}%${timeout}`;
}
